### PR TITLE
Fix rotation spinner snap value

### PR DESCRIPTION
**Pull request by ad-daniel**
**Description**
Fixes https://github.com/cyberbotics/webots/issues/4097.

For values around ~3.27, the snapping mechanism associated to the rotation spinner drops to zero (because near these values the `intpart` is 0), hence while rotating using the spinner arrows occasionally it can snap to 0 instead of decreasing/increasing at a constant rate (decreasing from 6.4 it snaps to PI for instance).
The proposed solution maintains the snapping behavior (at multiples of PI) but also the constant increase/decrease.

PS: axis-angle fields are restricted to [-PI, PI], so in practice putting values above PI will wrap them around anyway but having it jump when using the spinner arrows doesn't make sense either way.

### CODE REVIEW

File: src/webots/scene_tree/WbFieldDoubleSpinBox.cpp [hunk 1]
```diff
@@ -66,12 +66,12 @@ void WbFieldDoubleSpinBox::stepBy(int steps) {
       value += pow(10, -mDecimals) * steps;
       break;
     case RADIANS: {
-      // angle in radians: change by pi/24 and snap to pi/24 if close by 1%
+      // angle in radians: change by pi/24 and snap to the nearest multiple of PI
       value += M_PI * steps / 24.0;
       double intpart;
       double decpart = modf(fabs(value) / M_PI, &intpart);
```
Comment (BenjaminDeleze) [inline src/webots/scene_tree/WbFieldDoubleSpinBox.cpp:72]: Would it be possible to get rid of intpart, as it seems not used anymore?

**Review by BenjaminDeleze — COMMENTED** [review]
Seems good, just a question

File: src/webots/scene_tree/WbFieldDoubleSpinBox.cpp (hunk 1 above)
Comment (ad-daniel) [reply src/webots/scene_tree/WbFieldDoubleSpinBox.cpp]: indeed we can!

**Review by BenjaminDeleze — APPROVED** [review]
Thank you!

**Pull request merged**